### DISCUSSION TITLE
如何提高 ArrayTable 的性能

### DISCUSSION THREAD

**Discussion by JE-lee** [alibaba/formily#3098] [Q&A 2.x] [answered]
以 @formily/element 的 官网 Markup schema demo 为例，关掉了 vuedevtool 和 formily 插件的情况下, 一个 5 columns 的 ArrayTable 增加一行数据需要耗时 大概 80 ~ 90 ms，
我的机器是 win10 cpu 2.1g 4 core。

![image](https://user-images.githubusercontent.com/19794813/167083685-13808b86-f53d-4001-a3bb-eeb6a60186d4.png)
![image](https://user-images.githubusercontent.com/19794813/167083819-55887ac7-bbe0-404a-87b4-b4fcdb7708e0.png)

如果 table clolumns 数量比较多的话，耗时肯定更久。

虽然在大多数场景可以忍受，但是依然可以讨论下如何优化。

**Comment by janryWang** [comment, marked as answer]
如果真要极限优化，直接定制一个高性能table组件就行了，局部脱离formily，table列内容不用formily渲染，全是配置，组件自己控制列内容的显示
